Cython/Compiler/TreePath.py

```diff
@@ -167,6 +167,11 @@ def select(result):
                     continue
                 if attr_value == value:
                     yield attr_value
+                elif (isinstance(attr_value, bytes) and isinstance(value, str) and
+                        attr_value.decode("utf-8") == value):
```

Comment:
Shouldn't you encode `value` instead of decoding `attr_value`? (And only if it's `unicode` and not just `str`?)
How do you know that `attr_value` is really encoded in UTF-8 and not something else?